### PR TITLE
Updates rust version to 1.67.1

### LINKED ISSUE #1625: Update to use Rust 1.67.1 - https://github.com/rust-lang/rust/blob/master/RELEASES.md

### PR DESCRIPTION
**Pull request by betarelease**
## Description

Fixes pyrsia/pyrsia#1625

This PR updates rust version to 1.67.1 for 0.2.5

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/community/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've run [pre-commit.sh](https://github.com/pyrsia/pyrsia/blob/main/pre-commit.sh) ([pre-commit.bat](https://github.com/pyrsia/pyrsia/blob/main/pre-commit.bat)) successfully.
- [x] I've run the [integrations tests](https://github.com/pyrsia/pyrsia-integration-tests#how-to-set-up-and-run-the-tests)
on the PR branch and all tests passes.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ description = "Pyrsia Node and CLI"
 authors = ["pyrsiaoss <pyrsiaopensource@gmail.com>"]
 license = "Apache-2"
 repository = "https://github.com/pyrsia/pyrsia"
-rust-version = "1.66.1"
+rust-version = "1.67.1"
```
Comment (dasmanas) [inline Cargo.toml:10]: Should we also consider following 3 files to update rust version?

1. Dockerfile
2. installers/docker/CodeCoverageTest.Dockerfile
3. installers/docker/local_code_coverage/Dockerfile

**Review by dasmanas — DISMISSED** [review]

**Review by fishseabowl — DISMISSED** [review]

File: Cargo.toml (hunk 1 above)
Comment (betarelease) [reply Cargo.toml:10]: Updating the PR with these locations.

**Review by dasmanas — DISMISSED** [review]
lgtm

File: Cargo.toml (hunk 1 above)
Comment (sbtaylor15) [reply Cargo.toml:10]: It is not required to update the docker base images since rustup is run to set the correct version.

**Review by sbtaylor15 — DISMISSED** [review]
/lgtm

**Review by fishseabowl — DISMISSED** [review]

**Review by dasmanas — APPROVED** [review]
lg

**Review by fishseabowl — APPROVED** [review]

**Pull request merged**